mm2src/mm2_main/src/rpc/lp_commands/lp_commands_legacy.rs

```diff
@@ -45,37 +48,76 @@ pub async fn disable_coin(ctx: MmArc, req: Json) -> Result<Response<Vec<u8>>, St
         Ok(None) => return ERR!("No such coin: {}", ticker),
         Err(err) => return ERR!("!lp_coinfind({}): ", err),
     };
-    let swaps = try_s!(active_swaps_using_coin(&ctx, &ticker));
-    if !swaps.is_empty() {
-        let err = json!({
-            "error": format!("There're active swaps using {}", ticker),
-            "swaps": swaps,
-        });
-        return Response::builder()
-            .status(500)
-            .body(json::to_vec(&err).unwrap())
-            .map_err(|e| ERRL!("{}", e));
+    let coins_ctx = CoinsContext::from_ctx(&ctx).map_err(|err| ERRL!("{}", err))?;
+
+    // If a platform coin is to be disabled, we get all the enabled tokens for this platform coin first.
+    let mut coins_to_disable = coins_ctx.get_tokens_to_disable(&ticker).await;
+    // We then add the platform coin to the end of the list of the coins to be disabled.
+    coins_to_disable.push(ticker.clone());
+    drop_mutability!(coins_to_disable);
+
+    // Get all matching orders and active swaps.
+    let mut active_swaps = vec![];
+    let mut still_matching_orders = vec![];
+    for ticker in &coins_to_disable {
+        let swaps = try_s!(active_swaps_using_coin(&ctx, ticker));
+        let still_matching = try_s!(get_matching_orders(&ctx, ticker).await);
```

Comment:
This is inefficient due to the fact that `active_swaps_using_coin` and `get_matching_orders` may lock multiple mutexes and iterate over all orders and ongoing swaps.
Please consider refactoring them this way:
```rust
pub fn active_swaps_using_coins(ctx: &MmArc, coins: &HashSet<String>) -> Result<Vec<Uuid>, String> {}

pub async fn get_matching_orders(ctx: &MmArc, coins: &HashSet<String>) -> Result<Vec<Uuid>, String> {}
```

Reply:
done